src/test/java/com/uid2/admin/vertx/test/ServiceTestBase.java

```diff
@@ -166,6 +173,18 @@ protected void setKeysetKeys(int maxKeyId, KeysetKey... keys) throws Exception {
         });
     }
 
+    protected void setKeypairs(List<ClientSideKeypair> keypairs) throws Exception {
+        JsonObject metadata = new JsonObject();
+        HashMap<String, ClientSideKeypair> keypairMap = new HashMap<>();
+        keypairs.forEach(k -> keypairMap.put(k.getSubscriptionId(), k));
+        when(keypairProvider.getMetadata()).thenReturn(metadata);
+        when(keypairProvider.getAll()).thenReturn(keypairMap.values());
+        when(keypairProvider.getSnapshot().getAll()).thenReturn(new ArrayList<>(keypairMap.values()));
+        when(keypairSnapshot.getKeypair((anyString()))).thenAnswer(i -> {
```

Comment:
do we have to mock keypairSnapshot? can we not rely on keypairProvider.getSnapshot() ?

Reply:
Changed mock snapshot to real snapshot